fetch/src/main/scala/fetch.scala

```diff
@@ -377,7 +376,7 @@ object `package` {
         for {
           deferred <- Deferred[F, FetchStatus]
           request        = FetchOne(id, ds.data)
-          result         = deferred.complete _
+          result         = completeOrFail(deferred) _
```

Comment:
`Deferred#complete` now returns `F[Boolean]` instead of failing, so I've restored the "or fail" behavior with a helper